src/moj/components/sortable-table/sortable-table.mjs

```diff
@@ -25,71 +28,93 @@ SortableTable.prototype.setupOptions = function (params) {
 }
 
 SortableTable.prototype.createHeadingButtons = function () {
-  const headings = this.table.find('thead th')
-  let heading
-  for (let i = 0; i < headings.length; i++) {
-    heading = $(headings[i])
-    if (heading.attr('aria-sort')) {
-      this.createHeadingButton(heading, i)
+  for (const heading of this.headings) {
+    if (heading.hasAttribute('aria-sort')) {
+      this.createHeadingButton(heading)
     }
   }
 }
 
-SortableTable.prototype.createHeadingButton = function (heading, i) {
-  const text = heading.text()
-  const button = $(`<button type="button" data-index="${i}">${text}</button>`)
-  heading.text('')
-  heading.append(button)
+SortableTable.prototype.createHeadingButton = function (heading) {
+  const index = this.headings.indexOf(heading)
+  const button = document.createElement('button')
+
+  button.setAttribute('type', 'button')
+  button.setAttribute('data-index', index)
+  button.textContent = heading.textContent
+
+  heading.textContent = ''
+  heading.appendChild(button)
 }
 
 SortableTable.prototype.createStatusBox = function () {
-  this.status = $(
-    '<div aria-live="polite" role="status" aria-atomic="true" class="govuk-visually-hidden" />'
-  )
-  this.table.parent().append(this.status)
+  this.status = document.createElement('div')
+
+  this.status.setAttribute('aria-atomic', 'true')
+  this.status.setAttribute('aria-live', 'polite')
+  this.status.setAttribute('class', 'govuk-visually-hidden')
+  this.status.setAttribute('role', 'status')
+
+  this.table.insertAdjacentElement('afterend', this.status)
 }
 
 SortableTable.prototype.initialiseSortedColumn = function () {
-  const rows = this.getTableRowsArray()
-
-  this.table
-    .find('th')
-    .filter('[aria-sort="ascending"], [aria-sort="descending"]')
-    .first()
-    .each((index, el) => {
-      const sortDirection = $(el).attr('aria-sort')
-      const columnNumber = $(el).find('button').attr('data-index')
-      const sortedRows = this.sort(rows, columnNumber, sortDirection)
+  for (const heading of this.headings) {
+    const button = heading.querySelector('button')
+    const sortDirection = heading.getAttribute('aria-sort')
+
+    if (sortDirection === 'ascending' || sortDirection === 'descending') {
+      const columnNumber = button.getAttribute('data-index')
+      const sortedRows = this.sort(columnNumber, sortDirection)
+
       this.addRows(sortedRows)
```

Comment:
The jquery version has a `.first()` so it only applies an initial sort to the first heading it finds with a direction.  The new version goes through all headings applying a sort to each, so in a table with multiple columns having an initial sort direction applied, the new behaviour would be different - as it would be the last column that was sorted, not the first.

We probably want to break out of the `for` loop once we find a heading with a `sortDirection`